### DISCUSSION TITLE
Colour Support

### DISCUSSION THREAD

**Discussion by eBellmer** [vintlabs/fauxmoESP#178] [Q&A]
I've seen fauxmoESP advertised as having colour and dimming support, is this the case?
The master branch doesn't have any example code relating to colour, and the functions used in the examples from the 'colour' branch don't work with the master branch implying the functionality hasn't been merged yet. 

The 'colour' branch is effected by the discovery issue fixed in 3.3, and so doesn't seem to work anymore. 
Has colour support been discontinued?

**Comment by pvint** [comment]
Well, colour support was never a released feature, really. It was working nicely, however the way I implemented it breaks backwards compatibility, so I decided to roll it back and revisit the way it was handled. I've been a bit slow on getting back to it, but I expect I will soon.

As for dimming - that should work just fine! If you take the `value` in the callback `    fauxmo.onSetState([](unsigned char device_id, const char * device_name, bool state, unsigned char value) {` it gives you an 8 bit value to use for PWM.
